features/events/collector/src/main/java/org/opennms/netmgt/collection/EventMetricsCollector.java

```diff
@@ -0,0 +1,296 @@
+                                                    String groupName, CollectionGroup.Collection collection, IParm parm) {
+        Objects.requireNonNull(collection);
+        Objects.requireNonNull(parm);
+        String value = convertParamValue(collection, parm);
+        switch (collection.getType()) {
+            case GAUGE:
+                return collectionSet.withGauge(resource, groupName,
+                        this.getEscapeParamName(collection.getName()), Double.parseDouble(value));
+            case COUNTER:
+                return collectionSet.withCounter(resource, groupName,
+                        this.getEscapeParamName(collection.getName()), Double.parseDouble(value));
+            case STRING:
+                return collectionSet.withStringAttribute(resource, groupName,
+                        this.getEscapeParamName(collection.getName()), value);
+        }
+        return collectionSet;
+    }
+
+    private String getEscapeParamName(String name) {
+        return name.replaceAll(Matcher.quoteReplacement(File.separator), "_");
+    }
+
+    /**
+     * It will convert param values base on the eventconf's paramValues mapping. If nothing match, it returns the original value.
+     *
+     * @param collection
+     * @param parm
+     * @return converted value
+     */
+    private String convertParamValue(CollectionGroup.Collection collection, IParm parm) {
+        Objects.requireNonNull(parm);
+        Objects.requireNonNull(collection);
+        String value = parm.getValue().getContent();
+        String searchText = value + ":";
+        List<String> paramValues = collection.getParamValues();
+        for (var s : paramValues) {
+            var idx = s.indexOf(searchText);
+            if (idx != -1) {
+                return s.substring(idx + searchText.length());
+            }
+        }
+        return value;
+    }
+
+    private CollectionSetVisitor getRrdPersister(CollectionGroup.Rrd rrd) {
+        RrdRepository repository = new RrdRepository();
+        repository.setRrdBaseDir(rrd.getBaseDir());
+        repository.setStep(rrd.getStep());
+        repository.setHeartBeat(rrd.getHeartBeat());
+        repository.setRraList(rrd.getRras());
+        return persisterFactory.createPersister(
+                new ServiceParameters(Collections.emptyMap()), repository, false, false, false);
+    }
+
+    private IParm getMatchParam(IEvent e, CollectionGroup.Collection collection) {
+        Objects.requireNonNull(e);
+        Objects.requireNonNull(collection);
+        IParm parm = e.getParm(collection.getName());
+        if (parm != null){
+            return parm;
+        }
+        // if user set the collection name as uei, it means user want the whole message as value
+        if (e.getUei().equals(collection.getName()) && e.getParmCollection().size() == 1) {
+            return e.getParmCollection().get(0);
+        }
```

Comment:
I don't get this: if I use the uei as collection name, I would assume to match the message. This matches the only parameter and only if it really is the only one.

Reply:
rename added